### DISCUSSION TITLE
ability to splite code into compoents

### DISCUSSION THREAD

**Discussion by sohailahmad07** [alpinejs/alpine#4196] [3. Feature ideas]
hi can we get a feature in alpine js to split code into multiple components to get our code more organized i have an alpine components that are around 2000 lines it is kind of hard to go through that code. an ideal way to organize the code would be to create a main component. 
```

Alpine.component('whiteboard', ({
 //Import all the sub-components here so each file has other own responsibility  
 init() {
  //  init components
 }
 
 Alpine.load('add-tools')
 Alpine.load('upload-image')
 Alpine.load('change-properties')
 // etc

}));
```

**Comment by ekwoka** [comment]
I'm not sure exactly what this means.

Like you just have big files? Or?

What exactly?

You can split your files up however you want, I'm just not even sure what the issue is you're running into.

Like those just look like different components.


So you can just use them as different components...

**Comment by sohailahmad07** [comment]
ok so i have build a simple whiteboard tool and i am using fabric js 

my code simplified 

View
```
    <div x-data="whiteboard">
    //  toobar here
    
    // property bar here
    <!-- Canvas -->
        <canvas x-ref="drawingCanvas">
        </canvas>
    
    </div>
```


js
```
export default (function (options) {
    let useFabric;
    let canvas;
    
    return {
        stroke: this.$persist("rgba(0, 0, 0, 0)"),
        fill: this.$persist("red"),
        // other properties 
        
        // method here
        
        handleKeyDownEvent() {...}
        selectionTool(){...}
        changeFontStyle(){...}
        ... other method here
        }
```

is there a way i can put a similar feature method in different js files and import them here 


remember features are not completely isolated the need access to fabric and canvas variables and sometime event these other js file

**Reply by ekwoka** [reply]
put the data inside the component.

and you can import functions and add them to the object, yes.

**Reply by sohailahmad07** [reply]
@ekwoka how can you give me a simple example

**Reply by ekwoka** [reply]
Like a function?

```js

const keyDown = function() {
   console.log(this.stuff)
}

Alpine.data('thing', () => ({
  stuff: 'hi, mom',
  keyDown
})
```
